src/SFA.DAS.EmployerIncentives.Queries/Decorators/QueryHandlerWithLogging.cs

```diff
@@ -28,22 +28,22 @@ public async Task<TResult> Handle(TQuery query, CancellationToken cancellationTo
             {
                 if (domainLog.OnProcessing == null)
```

Comment:
Opportunity to simplify this decorator to this ...


```
var domainLog = (query is ILogWriter) ? (query as ILogWriter).Log : new Log() { OnProcessing = () => "", OnProcessed = () => "", OnError = () => "" };

            try
            {
                _log.LogDebug($"Start handle '{typeof(TQuery)}' query : {domainLog.OnProcessing.Invoke()}");
                var result = await _handler.Handle(query, cancellationToken);

                var responseLog = (result is ILogWriter) ? (result as ILogWriter).Log : new Log() { OnProcessed = () => "" };
                _log.LogDebug($"End handle '{typeof(TQuery)}' query : {domainLog.OnProcessed.Invoke()}, response : {responseLog.OnProcessed.Invoke()}");

                return result;
            }
            catch (Exception ex)
            {
                _log.LogError(ex, $"Error handling '{typeof(TQuery)}' query : {domainLog.OnError.Invoke()}");

                throw;
            }
```

Reply:
The above can be improved since it will throw error if one of the Log delegate methods isn't implemented  So to get round that initialise in the Log constructor ...

``` e.g.
 public Log()
        {
            OnProcessing = () => "";
            OnProcessed = () => "";
            OnError = () => "";
            OnProcessedWithArgs = () => new Tuple<string, object[]>("", null);
        }

```

Then the Log initialisation in the example above can be removed and non implemented delegates won't throw an error.